packages/api/tests/endpoints/test-granules-bulkMoveCollection.js

```diff
@@ -0,0 +1,336 @@
+    testContext: { knex, sfnMethod },
+    body: {
+      sourceCollectionId: t.context.collectionId,
+      targetCollectionId: t.context.collectionId2,
+      ...testBodyValues,
+    },
+  }, response);
+  const expected = {
+    execution: 'fakeArn',
+    message: `Successfully submitted bulk granule move collection with ${granuleIds.length} granules`,
+  };
+
+  const expectedPayload = {
+    cumulus_meta: {
+      state_machine: workflowArn,
+      cumulus_version: 'fakeVersion',
+    },
+    meta: {
+      bulkMoveCollection: {
+        ...testBodyValues,
+        targetCollection: deconstructCollectionId(t.context.collectionId2),
+      },
+    },
+    payload: {
+      granuleIds: granuleIds.sort(),
+    },
+    collection: {
+      name: t.context.collectionName,
+      version: t.context.collectionVersion,
+    },
+  };
+
+  const actualPayload = JSON.parse(startExecutionStub.getCall(0).args[0].input);
+  t.like(actualPayload, expectedPayload, 'StartExecution should match');
+  t.truthy(actualPayload.cumulus_meta.execution_name);
+  t.truthy(actualPayload.cumulus_meta.workflow_start_time);
+  t.deepEqual(response.send.getCall(0).args[0], expected, 'Endpoint response should match');
+});
+
+test.serial('bulkMoveCollection handles ExecutionAlreadyExists error correctly', async (t) => {
+  const { knex, workflowArn } = t.context;
+  const req = {
+    body: {
+      batchSize: 100,
+      concurrency: 10,
+      invalidBehavior: 'error',
+      sourceCollectionId: t.context.collectionId,
+      targetCollectionId: t.context.collectionId2,
+    },
+    testContext: {
+      knex,
+      sfnMethod: () => ({
+        startExecution: sinon.stub().throws(new ExecutionAlreadyExists('Execution already exists')),
+      }),
+    },
+  };
+
+  const res = {
+    boom: {
+      badRequest: sinon.stub(),
+    },
+    send: sinon.stub(),
+  };
+
+  await bulkMoveCollection(req, res);
+  t.true(res.boom.badRequest.firstCall.args[0].includes(`already exists for state machine ${workflowArn}`));
+});
+
+test.serial('bulkMoveCollection handles a missing workflow configuration file correctly', async (t) => {
+  const { knex } = t.context;
+  const startExecutionStub = sinon.stub();
+  startExecutionStub.returns({
+    executionArn: 'fakeArn',
+  });
+  const sfnMethod = () => ({
+    startExecution: startExecutionStub,
+  });
+  const req = {
+    body: {
+      batchSize: 100,
+      concurrency: 10,
+      invalidBehavior: 'error',
+      sourceCollectionId: t.context.collectionId,
+      targetCollectionId: t.context.collectionId2,
+    },
+    testContext: {
+      knex,
+      sfnMethod,
+      workflow: 'someDumbWorkflow',
+    },
+  };
+  const res = {
+    boom: {
+      badRequest: sinon.stub(),
+    },
+    send: sinon.stub(),
+  };
+  await bulkMoveCollection(req, res);
+  t.is('Unable to find state machine ARN for workflow someDumbWorkflow', res.boom.badRequest.firstCall.args[0], 'Error message should match catch for getJsonS3Object');
+});
```

Comment:
What about a test for a Collection with no Granules?

Reply:
Seems reasonable enough.  054cfa1a8fe - may refactor/DRY that up a bit, but should cover the thought.